scripts/trigger_builds.py

```
@@ -137,6 +137,24 @@ def monitor_builds(client: Client, builds: list[dict], poll_interval: float) ->
     return all(b["state"] == "succeeded" for b in builds)
 
 
+def positive_number(value: str | int | float) -> int | float:
+    message = f"Value must be a positive int or float: {value}"
+    try:
+        value = int(value)
+    except ValueError:
+        try:
+            value = float(value)
+        except ValueError:
+            raise argparse.ArgumentTypeError(message)
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)